tools/run_tests/report_utils.py

```diff
@@ -69,147 +74,36 @@ def render_xml_report(resultset, xml_report):
-  if http2_interop:
-    # Each column header is the server language.
-    html_str = ('%s<h2>HTTP/2 Interop</h2>\n' 
-                '<table style=\"width:100%%\" border=\"1\">\n'
-                '<tr bgcolor=\"#00BFFF\">\n'
-                '<th>Servers &#9658;<br/>'
-                'Test Cases &#9660;</th>\n') % html_str
-    for server_lang in sorted_server_langs:
-      html_str = '%s<th>%s\n' % (html_str, server_lang)
-    if cloud_to_prod:
-      html_str = '%s<th>%s\n' % (html_str, "prod")
-    html_str = '%s</tr>\n' % html_str
-    for test_case in sorted_http2_cases:
-      html_str = '%s<tr><td><b>%s</b></td>\n' % (html_str, test_case)
-      # Fill up the cells with test result.
-      for server_lang in sorted_server_langs:
-        shortname = 'cloud_to_cloud:%s:%s_server:%s' % (
-            "http2", server_lang, test_case)
-        html_str = fill_one_test_result(shortname, resultset, html_str)
-      if cloud_to_prod:
-        shortname = 'cloud_to_prod:%s:%s' % ("http2", test_case)
-        html_str = fill_one_test_result(shortname, resultset, html_str)
-      html_str = '%s</tr>\n' % html_str
-    html_str = '%s</table>\n' % html_str
-  if server_langs:
-    for test_case in sorted_test_cases:
-      # Each column header is the client language.
-      html_str = ('%s<h2>%s</h2>\n' 
-                  '<table style=\"width:100%%\" border=\"1\">\n'
-                  '<tr bgcolor=\"#00BFFF\">\n'
-                  '<th>Client languages &#9658;<br/>'
-                  'Server languages &#9660;</th>\n') % (html_str, test_case)
-      for client_lang in sorted_client_langs:
-        html_str = '%s<th>%s\n' % (html_str, client_lang)
-      html_str = '%s</tr>\n' % html_str
-      # Each row head is the server language.
-      for server_lang in sorted_server_langs:
-        html_str = '%s<tr><td><b>%s</b></td>\n' % (html_str, server_lang)
-        # Fill up the cells with test result.
-        for client_lang in sorted_client_langs:
-          shortname = 'cloud_to_cloud:%s:%s_server:%s' % (
-              client_lang, server_lang, test_case)
-          html_str = fill_one_test_result(shortname, resultset, html_str)
-        html_str = '%s</tr>\n' % html_str
-      html_str = '%s</table>\n' % html_str
 
-  html_str = ('%s\n'
-              '<script>\n'
-              '$(document).ready(function(){'
-              '$(\'[data-toggle=\"tooltip\"]\').tooltip();\n'   
-              '});\n'
-              '</script>\n'
-              '</body>\n'
-              '</html>') % html_str  
-  
-  # Write to reports/index.html as set up in Jenkins plugin.
-  html_report_dir = 'reports'
-  if not os.path.exists(html_report_dir):
-    os.mkdir(html_report_dir)
-  html_file_path = os.path.join(html_report_dir, 'index.html')
-  with open(html_file_path, 'w') as f:
-    f.write(html_str)
+  template_file = 'templates/interop_html_report.template'
+  mytemplate = Template(filename=template_file, format_exceptions=True)
```

Comment:
Won't this also fail on any machine that doesn't have mako installed?  I ran these tests manually on my dev machine so I would imagine other people have the use case.  

I don't believe this code is correct either.  If Template does not exist, the if statement will never be reached since an exception is thrown.  I also don't think the Template constructor will return a False-y object, so it would never be hit in the good case either.